### PR TITLE
Add enabled key to sections

### PR DESCRIPTION
**Pull request by JamesGardiner**
### PR Context
This PR proposes a new `is_enabled` key for section blocks - this will allow us to remove sections from the routing path based on answers.

### CODE REVIEW

**Comment by JamesGardiner** [comment]
[propose updated schema to support hub enabled based on radio or checkboxes](https://trello.com/c/kpAwSxeQ/3512-propose-updated-schema-to-support-hub-enabled-based-on-radio-or-checkboxes)

File: schemas/questionnaire_v1.json [hunk 1]
```diff
@@ -255,18 +239,27 @@
...
               "additionalProperties": false,
-              "required": [
-                "id",
-                "title",
-                "blocks"
-              ]
+              "required": ["id", "title", "blocks"]
             }
+          },
+          "is_enabled": {
```
Comment (iwootten) [inline schemas/questionnaire_v1.json]: The hub already uses the key "enabled" for the same behaviour. We should use the same key for both to ensure consistency.

File: schemas/questionnaire_v1.json [hunk 2]
```diff
@@ -64,13 +64,7 @@
...
           "type": {
             "type": "string",
-            "enum": [
-              "boolean",
-              "date",
-              "string",
-              "uuid",
-              "url"
-            ]
+            "enum": ["boolean", "date", "string", "uuid", "url"]
```
Comment (ajmaddaford) [inline schemas/questionnaire_v1.json]: The changes look good, but it looks like other reformatting has been done.

File: schemas/questionnaire_v1.json [hunk 3]
```diff
@@ -255,18 +239,27 @@
...
               "additionalProperties": false,
-              "required": [
-                "id",
-                "title",
-                "blocks"
-              ]
+              "required": ["id", "title", "blocks"]
             }
+          },
+          "enabled": {
```
Comment (MebinAbraham) [inline schemas/questionnaire_v1.json]: I guess it might be overkill but `enabled` here is dependant on the `hub` being `enabled`.

**Comment by MebinAbraham** [comment]
@JamesGardiner Can you update the PR title.

File: schemas/questionnaire_v1.json (hunk 3 above)
Comment (MebinAbraham) [reply schemas/questionnaire_v1.json]: Discussed with @ajmaddaford that they are independent and they can be used in cases without the notion of a hub.

File: schemas/questionnaire_v1.json (hunk 2 above)
Comment (MebinAbraham) [reply schemas/questionnaire_v1.json]: https://github.com/ONSdigital/eq-schema-validator/pull/171

**Comment by JamesGardiner** [comment]
Waiting for #171 to merge to v3

File: schemas/questionnaire_v1.json (hunk 2 above)
Comment (MebinAbraham) [reply schemas/questionnaire_v1.json]: https://github.com/ONSdigital/eq-schema-validator/pull/173

😁

File: schemas/questionnaire_v1.json [hunk 4]
```diff
@@ -238,6 +238,22 @@
               "additionalProperties": false,
               "required": ["id", "title", "blocks"]
             }
+          },
+          "enabled": {
+            "description": "Allows removing a section from the routing path based on when rules. By adding more than one `when` element it will evaluate as an or rule.",
```
Comment (MebinAbraham) [inline schemas/questionnaire_v1.json]: Can we adjust this description please as it is not quite right?

**Pull request merged**